src/Lean/Meta/Tactic/Grind/EMatchTheorem.lean

```diff
@@ -571,7 +571,8 @@ def mkEMatchTheoremCore (origin : Origin) (levelParams : Array Name) (numParams
 
 private def getProofFor (declName : Name) : MetaM Expr := do
   let info ← getConstInfo declName
-  unless info.isTheorem do
+  -- For theorems, `isProp` has already been checked at declaration time
+  unless wasOriginallyTheorem (← getEnv) declName do
```

Comment:
@leodemoura FYI - I'm assuming this was the intention of the check and that it is still helpful even if it requires an additional environment extension lookup. Also please let me know if you can think of a better name for the function!